src/core/random.hpp

```diff
@@ -101,38 +101,31 @@ Utils::Vector<uint64_t, 4> philox_4_uint64s(uint64_t counter, int key1,
  * not correlated between two calls along the same counter
  * sequence.
  *
- */
-template <RNGSalt salt> double noise(uint64_t counter, int key1, int key2 = 0) {
-
-  auto const noise = philox_4_uint64s<salt>(counter, key1, key2);
-
-  using Utils::uniform;
-  return uniform(noise[0]) - 0.5;
-}
-
-/**
- * @brief 3d uniform vector noise.
- *
- * Mean = 0, variance = 1 / 12.
- * This uses the Philox PRNG, the state is controlled
- * by the counter, the salt and two keys.
- * If any of the keys and salt differ, the noise is
- * not correlated between two calls along the same counter
- * sequence.
+ * @tparam salt RNG salt
+ * @tparam N    Size of the noise vector
  *
+ * @return Vector of uniform random numbers.
  */
-template <RNGSalt salt>
-Utils::Vector3d v_noise(uint64_t counter, int key1, int key2 = 0) {
+template <RNGSalt salt, size_t N = 3,
+          typename std::enable_if<(N > 1) and (N <= 4), int>::type = 0>
```

Comment:
`std::enable_if_t`. Also I prefer `class = std::enable_if_t...` because I find' the version without the extra `int` less noisy, but I guess this is a matter of taste.

Reply:
thanks for the suggestion